plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/TestingThriftHttpMetastoreServer.java

```diff
@@ -75,12 +75,18 @@ public TestingThriftHttpMetastoreServer(ThriftMetastore delegate, Consumer<HttpS
     private static <T> T proxyHandler(ThriftMetastore delegate, Class<T> iface)
     {
         return newProxy(iface, (proxy, method, args) -> switch (method.getName()) {
-            case "getAllDatabases" -> delegate.getAllDatabases();
+            case "getDatabases" -> delegate.getAllDatabases();
```

Comment:
Shared with @osscm a draft test case to get him started

```
package io.trino.plugin.hive.metastore;/*
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import com.google.common.collect.ImmutableMap;
import io.trino.Session;
import io.trino.plugin.hive.HiveQueryRunner;
import io.trino.plugin.hive.containers.HiveHadoop;
import io.trino.testing.AbstractTestQueryFramework;
import io.trino.testing.QueryRunner;
import org.junit.jupiter.api.AfterAll;
import org.junit.jupiter.api.Test;

import java.util.Optional;

import static io.trino.testing.TestingSession.testSessionBuilder;
import static org.assertj.core.api.Assertions.assertThat;

public class TestHiveMetastoreCatalogs
        extends AbstractTestQueryFramework
{
    private static final String TRINO_HIVE_CUSTOM_CATALOG = "hive_custom_catalog";
    private static final String HIVE_CUSTOM_CATALOG = "custom";
    private static final String HIVE_CUSTOM_SCHEMA = "custom_schema";

    private static final Session DEFAULT_SESSION = testSessionBuilder()
            .setCatalog("hive")
            .setSchema(Optional.empty())
            .build();
    private static final Session CUSTOM_SESSION = testSessionBuilder()
            .setCatalog(TRINO_HIVE_CUSTOM_CATALOG)
            .setSchema(HIVE_CUSTOM_SCHEMA)
            .build();

    private HiveHadoop hiveHadoop;

    @Override
    protected QueryRunner createQueryRunner()
            throws Exception
    {

        hiveHadoop = HiveHadoop.builder().build();
        hiveHadoop.start();

        QueryRunner queryRunner = HiveQueryRunner.builder(DEFAULT_SESSION)
                .addHiveProperty("hive.metastore", "thrift")
                .addHiveProperty("hive.metastore.uri", hiveHadoop.getHiveMetastoreEndpoint().toString())
                .setCreateTpchSchemas(false)
                .build();

        hiveHadoop.runOnMetastore("INSERT INTO CTLGS VALUES (2, '%s', 'Custom catalog', 'hdfs://hadoop-master:9000/user/hive/warehouse/custom')".formatted(HIVE_CUSTOM_CATALOG));

        queryRunner.createCatalog(TRINO_HIVE_CUSTOM_CATALOG, "hive", ImmutableMap.<String, String>builder()
                .put("hive.metastore", "thrift")
                .put("hive.metastore.uri", hiveHadoop.getHiveMetastoreEndpoint().toString())
                .put("hive.metastore.catalog.name", "custom")
                .buildOrThrow());

        queryRunner.execute(
                testSessionBuilder()
                        .setCatalog(TRINO_HIVE_CUSTOM_CATALOG)
                        .build(),
                "CREATE SCHEMA " + HIVE_CUSTOM_SCHEMA);
        return queryRunner;
    }

    @AfterAll
    void afterAll()
    {
        hiveHadoop.stop();
    }

    @Test
    public void testShowSchemas()
    {
        assertThat(query(DEFAULT_SESSION, "SHOW SCHEMAS")).matches("VALUES VARCHAR 'default', VARCHAR 'information_schema'");
        assertThat(query(CUSTOM_SESSION, "SHOW SCHEMAS")).matches("VALUES VARCHAR '" + HIVE_CUSTOM_SCHEMA + "', VARCHAR 'information_schema'");
    }
}

```

Reply:
@osscm pls also create an own extension of `BaseConnectorSmokeTest` similar to `TestHiveConnectorSmokeTest` e.g. `TestHiveConnectorCustomCatalogSmokeTest`

This would be needed to ensure that all the operations work as expected even when dealing with custom metastores.

```
    @Override
    protected QueryRunner createQueryRunner()
            throws Exception
    {
        hiveHadoop = HiveHadoop.builder().build();
        hiveHadoop.start();

        hiveHadoop.runOnMetastore("INSERT INTO CTLGS VALUES (2, '%s', 'Custom catalog', 'hdfs://hadoop-master:9000/user/hive/warehouse/custom')".formatted(HIVE_CUSTOM_CATALOG));
        

        QueryRunner queryRunner = HiveQueryRunner.builder(testSessionBuilder()
                        .setCatalog(TRINO_HIVE_CUSTOM_CATALOG)
                        .setSchema("custom_schema")
                        .build())
                .addHiveProperty("hive.metastore", "thrift")
                .addHiveProperty("hive.metastore.uri", hiveHadoop.getHiveMetastoreEndpoint().toString())
                .setCreateTpchSchemas(false)
                .build();

        queryRunner.execute(
                testSessionBuilder()
                        .setCatalog(TRINO_HIVE_CUSTOM_CATALOG)
                        .build(),
                "CREATE SCHEMA custom_schema");
        
        return queryRunner;
    }
```

Reply:
@findinpath extension of `BaseConnectorSmokeTest` is failing for multiple test-cases with custom catalog.
As we were discussing, most probably doing some QueryRunner setup mistake.
Added the test case now, so that we discuss, thanks!

Reply:
@findinpath please see, added the one you have asked and shared, using the `minio` instead of `hdfs`, thanks!